src/main/java/seedu/address/model/person/Person.java

```diff
@@ -24,16 +24,18 @@ public class Person {
     // Data fields
     private final Address address;
     private final Set<Tag> tags = new HashSet<>();
+    private final Note note;
 
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, tags);
+    public Person(Name name, Phone phone, Email email, Address address, Note note, Set<Tag> tags) {
+        requireAllNonNull(name, phone, email, address, note, tags);
         this.name = name;
         this.phone = phone;
         this.email = email;
         this.address = address;
+        this.note = note;
         this.tags.addAll(tags);
     }
```

Comment:
Since create Person and edit Person will not involve "Note" creation, and here you just insert a dummy empty String, i would suggest that it might be a better idea not to alter the constructor. As you might observed that, you have to change a lot part of code due to change of constructor. If i were you, to reduce disruption due to change of constructor, i might directly handle Note note = new Note("") at declaration instead of handling in constructor. 

Either or, i believe it still working perfectly fine and all the code are well writen with good code quality. Great Job

Reply:
noted !! i will consider the alternative